extensions/common/vault/vault-gcp/src/test/java/org/eclipse/edc/vault/gcp/GcpSecretManagerVaultTest.java

```diff
@@ -0,0 +1,277 @@
+                                .setLocation(TEST_REGION)
+                                .build())
+                            .build())
+                        .build())
+                    .build();
+
+    private ArrayList<Character> validChars = new ArrayList<Character>();
+    private ArrayList<Character> invalidChars = new ArrayList<Character>();
+    private Random randGen = new Random();
+
+    private boolean isValidKey(String key) {
+        if (key.length() > MAX_KEY_LENGTH) {
+            return false;
+        }
+
+        for (int i = 0; i < key.length(); i++) {
+            var c = key.charAt(i);
+            if (!Character.isLetterOrDigit(c) && c != '_' && c != '-') {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private char getRandomValidChar() {
+        return validChars.get(randGen.nextInt(validChars.size()));
+    }
+
+    private char getRandomInvalidChar() {
+        return invalidChars.get(randGen.nextInt(invalidChars.size()));
+    }
+
+    private String getRandomKeyWithValidChars(int length) {
+        var sb = new StringBuilder();
+        for (int i = 0; i < length; i++) {
+            var c = getRandomValidChar();
+            sb.append(c);
+        }
+        return sb.toString();
+    }
+
+    private String getRandomKeyWithInvalidChars(int length, int errors) {
+        String key = getRandomKeyWithValidChars(length);
+        StringBuilder keyBuilder = new StringBuilder(key);
+        ArrayList<Integer> positions = new ArrayList<Integer>();
+        for (int i = 0; i < length; i++) {
+            positions.add(i);
+        }
+
+        for (int i = 0; i < errors && positions.size() > 0; i++) {
+            int index = randGen.nextInt(positions.size());
+            int position = positions.get(index);
+            positions.remove(index);
+
+            keyBuilder.setCharAt(position, getRandomInvalidChar());
+        }
+
+        return keyBuilder.toString();
+    }
+
+    @BeforeAll
+    void init() {
+        // Init the array with the chars allowed and not allowed in a Secret Key.
+        // Secret key is a string with a maximum length of 255 characters and can
+        // contain uppercase and lowercase letters, digits, and the hyphen (`-`) and
+        // underscore (`_`) characters.
+        for (char c = 'a'; c <= 'z'; c++) {
+            validChars.add(c);
+        }
+        for (char c = 'A'; c <= 'Z'; c++) {
+            validChars.add(c);
+        }
+        for (char c = '0'; c <= '9'; c++) {
+            validChars.add(c);
+        }
+        validChars.add('-');
+        validChars.add('_');
+
+        for (char c = 32; c <= 126; c++) {
+            if (!Character.isLetterOrDigit(c) && c != '_' && c != '-') {
+                invalidChars.add(c);
+            }
+        }
+
+        randGen.setSeed(System.currentTimeMillis());
+    }
+
+    @BeforeEach
+    void resetMocks() {
+        reset(monitor, secretClient);
```

Comment:
pls don't re-use mocks, just reinstantiate them on every run -> consistency across the code base.